runbotics-orchestrator/src/main/java/com/runbotics/web/rest/ProcessCollectionResource.java

```diff
@@ -36,28 +38,57 @@ public class ProcessCollectionResource {
     private final ProcessCollectionQueryService processCollectionQueryService;
 
     private final ProcessCollectionRepository processCollectionRepository;
+
+    private final UserService userService;
+
     @Value("${jhipster.clientApp.name}")
     private String applicationName;
 
     public ProcessCollectionResource(
         ProcessCollectionService processCollectionService,
         ProcessCollectionQueryService processCollectionQueryService,
-        ProcessCollectionRepository processCollectionRepository
+        ProcessCollectionRepository processCollectionRepository,
+        UserService userService
     ) {
         this.processCollectionService = processCollectionService;
         this.processCollectionRepository = processCollectionRepository;
         this.processCollectionQueryService = processCollectionQueryService;
+        this.userService = userService;
+    }
+
+    @PreAuthorize("@securityService.checkFeatureKeyAccess('" + FeatureKeyConstants.PROCESS_COLLECTION_READ + "')")
+    @GetMapping("process-collection/access")
+    public ResponseEntity getCollectionAccess(ProcessCollectionCriteria criteria) {
+        User requester = userService.getUserWithAuthorities().get();
+
+        if (criteria.getParentId() != null) {
+            processCollectionService.checkCollectionAvailability(
+                criteria.getParentId().getEquals(), requester
+            );
+        }
+
+        return ResponseEntity.accepted().build();
     }
 
     @PreAuthorize("@securityService.checkFeatureKeyAccess('" + FeatureKeyConstants.PROCESS_COLLECTION_READ + "')")
     @GetMapping("process-collection")
     public ResponseEntity<List<ProcessCollectionDTO>> getAllCollections(ProcessCollectionCriteria criteria) {
-        List<ProcessCollectionDTO> collections = processCollectionService.getCollectionsByCriteria(criteria);
-
-        if (collections.isEmpty()) {
-            return ResponseEntity.notFound().build();
+        User requester = userService.getUserWithAuthorities().get();
+        boolean hasUserAccessEveryCollection = requester.getFeatureKeys().contains(FeatureKeyConstants.PROCESS_COLLECTION_EVERY);
+
+        List<ProcessCollectionDTO> result;
+        if (criteria.getParentId() != null) {
+            processCollectionService.checkCollectionAvailability(
+                criteria.getParentId().getEquals(), requester
+            );
+
+            result = processCollectionService.getChildrenCollectionsByParent(
+                criteria.getParentId().getEquals(), requester);
+        } else {
+            result = processCollectionService.getChildrenCollectionsByRoot(requester);
         }
-        return ResponseEntity.ok().body(collections);
+
+        return ResponseEntity.ok().body(result);
```

Comment:
I think it's better to return ResponseEntity.ok().body(result); twice but stick to functional programming and not reassigning result